File: GraffiTab_iOS_SDK/Classes/Model/Activity/Container/GTActivityContainer.m
//
//  GTActivityContainer.m
//  GraffiTab_iOS_SDK
//
//  Created by Georgi Christov on 08/05/2015.
//  Copyright (c) 2015 GraffiTab. All rights reserved.
//

#import "GTActivityContainer.h"

@implementation GTActivityContainer

- (id)initFromJson:(NSDictionary *)json {
    self = [super init];
    
    if (self) {
        self.activityUser = [[GTPerson alloc] initFromJson:json[JSON_ACTIVITY_CONTAINER_ACTIVITY_USER]];
        self.date = [NSDate dateWithTimeIntervalSince1970:[json[JSON_ACTIVITY_CONTAINER_DATE] longValue] / 1000];
        self.type = ActivityType(json[JSON_ACTIVITY_CONTAINER_TYPE]);
        self.activities = [NSMutableArray new];
        
        NSArray *jsonArray = json[JSON_ACTIVITY_CONTAINER_ACTIVITIES];
        for (NSDictionary *activityJson in jsonArray) {
            GTActivity *a = [GTActivityFactory createActivityFromJson:activityJson];
            [self.activities addObject:a];
        }
    }
    
    return self;
}

- (NSDictionary *)asDictionary {
    NSMutableDictionary *json = [NSMutableDictionary new];
    
    json[JSON_ACTIVITY_CONTAINER_ACTIVITY_USER] = self.activityUser.asDictionary;
    json[JSON_ACTIVITY_CONTAINER_DATE] = @(self.date.timeIntervalSince1970);
    json[JSON_ACTIVITY_CONTAINER_TYPE] = ACTIVITY_TYPE_FROM_OBJ_C[ACTIVITY_TYPE_LIST[self.type]];
    
    NSMutableArray *activitiesJson = [NSMutableArray new];
    for (GTActivity *a in self.activities)
        [activitiesJson addObject:a.asDictionary];
    
    json[JSON_ACTIVITY_CONTAINER_ACTIVITIES] = activitiesJson;
    
    return json;
}

@end

File: GraffiTab-iOS-SDK/Model/Activity/GTActivityContainer.swift
//
//  GTActivityContainer.swift
//  GraffiTab-iOS-SDK
//
//  Created by Georgi Christov on 11/04/2016.
//  Copyright © 2016 GraffiTab. All rights reserved.
//

import UIKit
import ObjectMapper

public class GTActivityContainer: Mappable {

    public var user: GTUser?
    public var date: NSDate?
    public var type: GTActivityType?
    public var activities: [GTActivity]?
    
    required public init?(_ map: Map) {
        
    }
    
    public func mapping(map: Map) {
        user <- map["user"]
        date <- (map["date"], CustomDateFormatTransform(formatString: GTApiDateConstants.InputFormat))
        type <- map["type"]
        activities <- map["activities"]
    }
}
